File: Assignment3/spl-net/src/main/java/bgu/spl/net/api/StompFrame.java
package bgu.spl.net.api;


import java.util.HashMap;

public class StompFrame {

    protected String command;
    protected HashMap<String, String> headers;
    protected String FrameBody;

    public StompFrame() {
        headers = new HashMap<>();
        FrameBody = "";
    }

    public StompFrame(String msg) {
        headers = new HashMap<>();
        String[] splitedArray = msg.split("\n");
        command = splitedArray[0];
        String[] headerRow;
        for( int i = 1; i < splitedArray.length; i++) {
            if(splitedArray[i].contains(":")) {
                headerRow = splitedArray[i].split(":");
                headers.put(headerRow[0], headerRow[1]);
            }
        }
        if(!splitedArray[splitedArray.length - 1].contains(":"))
            FrameBody = splitedArray[splitedArray.length - 1];
    }

    public String getCommand() {
        return command;
    }

    public String getFrameBody() {
        return FrameBody;
    }

    public void setCommand(String command) {
        this.command = command;
    }

    public void setFrameBody(String frameBody) {
        this.FrameBody = frameBody;
    }

    public HashMap<String, String> getHeaders() {
        return headers;
    }

    public void addHeader(String key, String value) {
        headers.put(key, value);
    }
}

File: Assignment3/Client/src/StompFrame.cpp
#include "../include/StompFrame.h"
#include <map>

using namespace std;

StompFrame::StompFrame() {
    command = "";
    headers = map<string, string>();
    frameBody = "";
}

const string &StompFrame::getCommand() const {
    return command;
}

const string &StompFrame::getFrameBody() const {
    return frameBody;
}

const map<string, string> &StompFrame::getHeaders() const {
    return headers;
}

void StompFrame::setCommand(const string &command) {
    StompFrame::command = command;
}

void StompFrame::setFrameBody(const string &frameBody) {
    StompFrame::frameBody = frameBody;
}

void StompFrame::addHeader(string str1, string str2) {
    StompFrame::headers.insert(pair<string,string>(str1, str2));
}

